pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -46,14 +46,73 @@ var (
 	}
 )
 
-// ApplyParameters applies the params from a TaskRun.Input.Parameters to a TaskSpec
-func ApplyParameters(ctx context.Context, spec *v1.TaskSpec, tr *v1.TaskRun, defaults ...v1.ParamSpec) *v1.TaskSpec {
+// applyStepActionParameters applies the params from a TaskRun.Input.Parameters to a TaskSpec
+func applyStepActionParameters(ctx context.Context, step *v1.Step, spec *v1.TaskSpec, tr *v1.TaskRun, stepParams v1.Params, defaults []v1.ParamSpec) *v1.Step {
```

Comment:
It looks like there is a lot of repetitive code with `ApplyParameters`, I think one followup is to cleanup the code if possible

Reply:
Or we can just do it in this PR.

Reply:
I recommend that any cleanup should happen in a separate PR that is merged before this one 🙏🏾

Reply:
I don't think that would be necessary since the cleanup is quite small and irrelevant without the need for step actions. PTAL. I'm happy to separate it into a different PR if you still think so.

Reply:
I was about to open that PR but saw that the changes have already been made in this one.  😄 If it is a separate we can also merge it very quickly.
https://github.com/Yongxuanzhang/pipeline/tree/refactor-taskrun-apply